### PR TITLE
LCH-3698 / LEB-4296: Fix block content rendering.

### PR DESCRIPTION
**Pull request by saxumVermes**

### CODE REVIEW

**Review by c0d39uru — APPROVED** [review]

**Review by c0d39uru — APPROVED** [review]

**Comment by scor** [comment]
Hi @saxumVermes, I wanted to merge your PR but looks like there are now some failures, not sure why they suddenly appeared since the tests were passing one month ago:
```
1) Drupal\Tests\acquia_lift_publisher\Kernel\EventSubscriber\EntityRenderHandlerTest::testImageAttributeIsSet
Error: Call to a member function getAttribute() on boolean
/home/travis/build/acquia/acquia_lift/modules/acquia_lift_publisher/tests/src/Kernel/EventSubscriber/Cdf/EntityRenderHandlerTest.php:202
--
1) Drupal\Tests\acquia_lift_publisher\Kernel\EventSubscriber\EntityRenderHandlerTest::testOnCreateCdfBlockContent
All entities were rendered.
Failed asserting that actual size 0 matches expected size 2.
/home/travis/build/acquia/acquia_lift/modules/acquia_lift_publisher/tests/src/Kernel/EventSubscriber/Cdf/EntityRenderHandlerTest.php:229
```
edit: Those errors appear despite making the 2 changes I mentioned below.

File: modules/acquia_lift_publisher/tests/src/Kernel/EventSubscriber/Cdf/EntityRenderHandlerTest.php [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+use Drupal\Tests\node\Traits\ContentTypeCreationTrait;
+use Drupal\Tests\node\Traits\NodeCreationTrait;
+use Drupal\Tests\RandomGeneratorTrait;
+use Drupal\Tests\TestFileCreationTrait;
+use Drupal\Tests\user\Traits\UserCreationTrait;
+use Prophecy\Argument;
+use Drupal\file\Entity\File;
+
+/**
+ * Class EntityRenderHandlerTest.
```
Comment (scor) [inline modules/acquia_lift_publisher/tests/src/Kernel/EventSubscriber/Cdf/EntityRenderHandlerTest.php:31]: we're missing a `@group acquia_lift` here.

File: modules/acquia_lift_publisher/tests/src/Kernel/EventSubscriber/Cdf/EntityRenderHandlerTest.php [hunk 2]
```diff
@@ -0,0 +1,380 @@
+<?php
+
+namespace Drupal\Tests\acquia_lift_publisher\Kernel\EventSubscriber;
```
Comment (scor) [inline modules/acquia_lift_publisher/tests/src/Kernel/EventSubscriber/Cdf/EntityRenderHandlerTest.php]: missing `\Cdf`, should read:
```
namespace Drupal\Tests\acquia_lift_publisher\Kernel\EventSubscriber\Cdf;
```

**Review by scor — CHANGES_REQUESTED** [review]

**Comment by scor** [comment]
I've replayed all the changes that happened on the 4.x branch since you posted the inital PR (initially based on top of 8.x-4.1) and I've narrowed it down to #288... something in this other PR about Publish only required entities introduced something that is not compatible with this PR.

**Review by scor — APPROVED** [review]

**Pull request merged**